test/lsp-completion-test.el

```diff
@@ -102,5 +102,32 @@
       (should (equal (do-test-trigger-kind t) 2))
       (should (equal (do-test-trigger-character t) "_")))))
 
+(ert-deftest lsp-completion-test-get-context-incomplete-completion ()
+  (setq lsp-completion--cache (list "not-used" :incomplete))
```

Comment:
I'm somewhat worried about this setting of `lsp-completion--cache` leaking out beyond the scope of this test. I think we may need to wrap this whole test in an unwind-protect so that regardless of whether the test passes or fails we set `lsp-completion--cache` back to nil. Thoughts?

Reply:
I don't think that investing time here will be worth the effort. The cache is ignored when completion is triggered from a different position, thus it is not very likely to pollute the other tests.